crypto/s2n_pkey.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "crypto/s2n_rsa.h"
+
+struct s2n_pkey {
+    union {
```

Comment:
Without going too much into how #539 was written, my plan was that `s2n_pkey` will also be initialized with function pointers to the key-type-specific functions. The key-type-specific functions are aware of which key should be populated in the union and are the only points in the code that actually operate on that union. I originally put a type in `s2n_pkey`, but realized at the end that the type field was never used so I removed it. I should also note that the type in `s2n_pkey` was only the type of the key itself, and was not the same as `s2n_cert_type` which represents a certificate type (related to both the key type and how the certificate was signed).